### DISCUSSION TITLE
Sauce Results Update

### DISCUSSION THREAD

**Discussion by Bajibabup** [webdriverio/webdriverio#9971] [General]
Hi,
I am using Sauce service "@wdio/sauce-service": "^8.3.9". Sauce connectivity and tests execution is good(even through Jenkins).However, in Sauce test execution status is being always updated as "Completed" than PASS or FAIL. 
Pls let me know if I need to configure anything in WDIO.conf.js, or to have any specific Sauce capabilities to have correct Sauce tests execution status.

**Comment by christian-bromann** [comment]
I just created a new project with `npm init wdio@latest ./` and selected running tests on Sauce with the service and it properly set the job name and status: https://app.saucelabs.com/tests/4ac8ea2954424d94a1b084bd42b3baa7?auth=cb08cdc9cd495616590ec639568ee027 .. some something must be wrong with your setup. Ensure that the service is set.

**Comment by Bajibabup** [comment]
Thx!
I have cross checked. Everything looks good. Here are the details. Jenkin pipeline has created , in the Jenkins execution stage/step sauce connection/tunnel will be initiated followed by the execution. Its picking the test suite and using Sauce service for execution, however only status is not being updated properly.  I have provided  info on dependences, wdio.conf.js info, and Jenkins pipeline syntax for execution. Pls glance and let me know.
Note: Requirement is to execute it through the only Jenkins

Dependences:
"devDependencies": {
          "@wdio/cli": "^8.3.3",
           "@wdio/local-runner": "^8.3.3",
        "@wdio/mocha-framework": "^8.3.0",
        "@wdio/sauce-service": "^8.3.9",
      },

Config File(I have kept only required info , removed all other properties)
exports.config = {
    user: process.env.SAUCE_USERNAME, 
    key:process.env.SAUCE_ACCESS_KEY, 
    region: "us-west",  
    
      
    // ============
    // Capabilities
    // ============
    
    maxInstances: 10,
 
    capabilities: [{
    
        maxInstances: 5,
        
        browserName: 'chrome',
        platformName: 'Windows 10',
         browserVersion: 'latest',
         acceptInsecureCerts: true,
        'sauce:options': {
            build: `Build ${Math.ceil(Date.now() / 1000)}`
                  },
        
        services: [
        ['sauce', {sauceConnect: true,}],  ],
         }

Jenkins Pipeline syntax for execution stage:

stage('Saucelabs test trigger') {
      steps {
        #It fetched the credentials from Jenkins credentials management
        sauce(credentialsId: 'bcred') {
         #Open the Sauce Connection
         sauceconnect(options: '--no-ssl-bump-domains *.deltadentalins.com,*.dentegra.com,*.deltadev.ent', 
 useGeneratedTunnelIdentifier: true, useLatestSauceConnect: true, verboseLogging: true) {
            sh '''
           echo "Sauce Connect is Success"
           cd ${WORKSPACE} 
            npm run ${TENVIRONMENT} -- --suite ${TESTSUITE} --service ${SERVICE} --browserName ${BROWSERNAME} 
                    
            '''
          }

**Reply by christian-bromann** [reply]
Your example is hard to read 😞 from what I see is that you put the `services` config into the `capabilities` object which is not allowed.

**Reply by Bajibabup** [reply]
I am sorry, Its a typo when I am copying from my config file. In-fact , its outside capabilities as below.


process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0"

exports.config = {
        user: process.env.SAUCE_USERNAME,
        key: process.env.SAUCE_ACCESS_KEY,
        region: "us-west",
        services:[
            ['sauce', {
            sauceConnect: true,
            }],
        ],
        capabilities: [{
            maxInstances: 5,
            acceptInsecureCerts: true,
            browserName: 'chrome',
            browserVersion: 'latest',
            platformName: 'Windows 10',
            'sauce:options': {
                //tunnelIdentifier: 'bajitest',
                build: `Build ${Math.ceil(Date.now() / 1000)}`
            },
        }],
        suites: {
            browserinstn:['test/specs/UnitTests/UIUnitTests/seleniumUnittests/browserInstantiation.js'],
            pomDatadriven:['test/specs/UnitTests/UIUnitTests/seleniumUnittests/pageObjectDatadriven.js'],
            asyncTest:['test/specs/UnitTests/UIUnitTests/seleniumUnittests/jasmineAsync.js']
                            
        },
    }

**Reply by christian-bromann** [reply]
I am not sure what to do , again it works with the boilerplate project. I recommend to do some debugging.

**Reply by Bajibabup** [reply]
Thanks Chris!
Let me try again. Just would like to check, do I need to provide any additional Sauce options in the capabilities?

**Reply by christian-bromann** [reply]
no

**Comment by charlesh21** [comment]
Hi All,
I had same results with Completed status on saucelabs dashboard.
Check with the process logs, I found the below errors, maybe cause the update test result fail

```
2023-03-20T15:48:32.377Z ERROR @wdio/utils:initialiseServices: Error: Couldn't initialise "@wdio/sauce-service".
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/coachh/git/prostorefe_wdio_uitest_js/node_modules/@wdio/sauce-service/build/index.js from /Users/coachh/git/prostorefe_wdio_uitest_js/node_modules/@wdio/utils/build/utils.js not supported.
Instead change the require of index.js in /Users/coachh/git/prostorefe_wdio_uitest_js/node_modules/@wdio/utils/build/utils.js to a dynamic import() which is available in all CommonJS modules.

```

**Comment by weagileyou** [comment]
Hi all, I had the same issue and I managed to fix it by changing/fixing hostname and region props:

before:
{
hostname: "ondemand.eu-central-1.saucelabs.com",
region: "us"
}

after:
{
region: "eu"
}
